src/core/mainloop/mainloop.c

```diff
@@ -1959,6 +1959,16 @@ write_stats_file_callback(time_t now, const or_options_t *options)
     if (next_write && next_write < next_time_to_write_stats_files)
       next_time_to_write_stats_files = next_write;
   }
+  /* Writes bandwidth statistics to disk if either it is set to 1
+   * or auto (-1). It doesn't check the consensus parameter here
+   * for it can change temporarily and we do not want to lose the stats.
+   * If BandwidthStatistics is exclusively set to 0 then don't write to
+   * disk. */
```

Comment:
These sentences could be a bit confusing:
```suggestion
  /* Writes bandwidth statistics to disk if BandwidthStatistics is set to 1
   * or auto (-1). It doesn't check the consensus parameter here
   * for it can change temporarily and we do not want to lose the stats.
   * If BandwidthStatistics is set to 0 then don't write to disk. */
```